Pipeline health page

ewels opened the pull request:
* [x] Page to display statuses
* [ ] Check all statuses
* [ ] Button to fix errors where possible

Closes https://github.com/nf-core/nf-co.re/issues/210

ewels commented:
Current state of play:

![localhost_23000_pipeline_health](https://user-images.githubusercontent.com/465550/67152221-0dcb6c00-f2d2-11e9-9992-d4c5ea408592.png)

Please don't fix the errors where they are present just yet - I need to be able to test something when I write the code to automatically fix it! 😆

ewels commented:
Also, although the code isn't doing a stupid number of API calls, it's still ridiculously slow. Takes about 30s to load. So need to optimise this quite a bit - probably by caching results and running on a cron job I guess. That's what I do everywhere else.

drpatelh commented on public_html/pipeline_health.php:
```diff
@@ -0,0 +1,281 @@
+<?php
+$title = 'Pipeline health';
```
```suggestion
$title = 'Repository health';
```

drpatelh commented on public_html/pipeline_health.php:
```diff
@@ -0,0 +1,281 @@
+<?php
+$title = 'Pipeline health';
+$subtitle = 'Check repository settings for all nf-core pipelines';
```
```suggestion
$subtitle = 'Check GitHub repository settings for all nf-core pipelines';
```

drpatelh commented:
So cool :+1: Thanks! Can we include the core repos in this too? I remember we missed some of these last time we imposed the merge restrictions to `dev` and `master`.

ewels commented:
Yup! Back end code is already written for that, just didn’t get around the printing the results. I even did nice OOP code with class inheritance so that we can easily add pipeline-specific or core-repo-specific tests in the future if we want to :sparkles:

ewels commented:
Updated screenshot:

![localhost_23000_pipeline_health (4)](https://user-images.githubusercontent.com/465550/67437869-3add8e80-f5f2-11e9-85cc-b8ea862f587e.png)

The table scrolls sideways and is now so wide that it won't fit on my screen for a screenshot 😆 At least, not unless I zoom out super far so you can't see anything:

![image](https://user-images.githubusercontent.com/465550/67437926-63658880-f5f2-11e9-883a-7406d0372137.png)

I _think_ that it now tests for everything mentioned in https://github.com/nf-core/nf-co.re/issues/210 .. though you might find some stuff that is reporting weirdly.

The table cells now have hover text to show you what they are for (useful when the table gets so big), and clicking on the table cell opens a new tab with the relevant repository settings page.

Finally, the caching is now implemented, so the page loads in a relatively normal time. However, if you click the magic blue button at the bottom it pulls everything from scratch again. This takes a long time, so I even made it so that you can pull a single repo - fewer API calls, so quite a bit quicker (pulling even one repo refreshes some tests for all repos though, such as team access). I did this after the screenshots above, so you'll have to just imagine a drop down box with repo names next to the blue button.

Ok - I think this is ready to be reviewed (?!?) and / or merged. The page doesn't actually _fix_ anything for you yet, but it does show all of the tests that we discussed which is a good starting point. I can do the code to fix the faults in a separate PR.

Phil

ewels commented:
Small improvement - I added an option to merge table column headings. So now there's just `Branch protection: master` and the mouse hover tooltips tell you what each specific test is for. I think it's better and it makes the table a lot more manageable:

![localhost_23000_pipeline_health (6)](https://user-images.githubusercontent.com/465550/67455471-fa036b00-f62d-11e9-8ca4-99f17c10a083.png)
